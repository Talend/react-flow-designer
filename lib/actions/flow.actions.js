'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.loadFlow = exports.resetFlow = exports.addFlowElements = undefined;
exports.setZoom = setZoom;

var _flowdesigner = require('../constants/flowdesigner.constants');

/**
 * Ask to sequentially add elements to the flow, each creation should be checked against store,
 * then applied via current reducers
 *
 * @params {array} listOfActionCreation
 */
var addFlowElements = exports.addFlowElements = function addFlowElements(listOfActionCreation) {
	return {
		type: _flowdesigner.FLOWDESIGNER_FLOW_ADD_ELEMENTS,
		listOfActionCreation: listOfActionCreation
	};
};

/**
 * ask for flow reset, emptying, nodes, links, ports collections
 */
var resetFlow = exports.resetFlow = function resetFlow() {
	return {
		type: _flowdesigner.FLOWDESIGNER_FLOW_RESET
	};
};

/**
 * reset old flow, load elements for the new flow
 */
var loadFlow = exports.loadFlow = function loadFlow(listOfActionCreation) {
	return {
		type: _flowdesigner.FLOWDESIGNER_FLOW_LOAD,
		listOfActionCreation: listOfActionCreation
	};
};

function setZoom(transform) {
	if (!isNaN(transform.k) && !isNaN(transform.x) && !isNaN(transform.y)) {
		return {
			type: _flowdesigner.FLOWDESIGNER_FLOW_SET_ZOOM,
			transform: transform
		};
	}
	return null;
}