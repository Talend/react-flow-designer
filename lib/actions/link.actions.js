'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.removeLink = exports.removeLinkData = exports.setLinkData = exports.removeLinkGraphicalAttribute = exports.setLinkGraphicalAttributes = exports.setLinkSource = exports.setLinkTarget = exports.addLink = undefined;

var _flowdesigner = require('../constants/flowdesigner.constants');

/**
 * Ask for link creation
 * @param {string} linkId
 * @param {string} sourceId - the source port Identifier
 * @param {string} targetId - the target port Identifier
 * @param {string} linkType
 * @param {Object} attr
 */
var addLink = exports.addLink = function addLink(linkId, sourceId, targetId) {
  var _ref = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : {},
      _ref$data = _ref.data,
      data = _ref$data === undefined ? {} : _ref$data,
      _ref$graphicalAttribu = _ref.graphicalAttributes,
      graphicalAttributes = _ref$graphicalAttribu === undefined ? {} : _ref$graphicalAttribu;

  return {
    type: _flowdesigner.FLOWDESIGNER_LINK_ADD,
    linkId: linkId,
    sourceId: sourceId,
    targetId: targetId,
    data: data,
    graphicalAttributes: graphicalAttributes
  };
};

/**
 * Ask for change of link target
 * @deprecated
 * @param {string} linkId
 * @param {string} targetId - the target port identifier
 */
var setLinkTarget = exports.setLinkTarget = function setLinkTarget(linkId, targetId) {
  return {
    type: _flowdesigner.FLOWDESIGNER_LINK_SET_TARGET,
    linkId: linkId,
    targetId: targetId
  };
};

/**
 * Ask for change of link source
 * @deprecated
 * @param {string} linkId
 * @param {string} sourceId - the source port identifier
 */
var setLinkSource = exports.setLinkSource = function setLinkSource(linkId, sourceId) {
  return {
    type: _flowdesigner.FLOWDESIGNER_LINK_SET_SOURCE,
    linkId: linkId,
    sourceId: sourceId
  };
};

/**
 * Ask to set graphical attributes on link
 * @deprecated
 * @param {string} linkId
 * @param {Object} attr
 */
var setLinkGraphicalAttributes = exports.setLinkGraphicalAttributes = function setLinkGraphicalAttributes(linkId, graphicalAttributes) {
  return {
    type: _flowdesigner.FLOWDESIGNER_LINK_SET_GRAPHICAL_ATTRIBUTES,
    linkId: linkId,
    graphicalAttributes: graphicalAttributes
  };
};

/**
 * Ask to remove an graphical attribute on target link
 * @deprecated
 * @param {string} linkId
 * @param {string} attrKey - the key of the attribute to be removed
 */
var removeLinkGraphicalAttribute = exports.removeLinkGraphicalAttribute = function removeLinkGraphicalAttribute(linkId, graphicalAttributesKey) {
  return {
    type: _flowdesigner.FLOWDESIGNER_LINK_REMOVE_GRAPHICAL_ATTRIBUTES,
    linkId: linkId,
    graphicalAttributesKey: graphicalAttributesKey
  };
};

/**
 * Ask to set data on link
 * @deprecated
 * @param {string} linkId
 * @param {Object} attr
 */
var setLinkData = exports.setLinkData = function setLinkData(linkId, data) {
  return {
    type: _flowdesigner.FLOWDESIGNER_LINK_SET_DATA,
    linkId: linkId,
    data: data
  };
};

/**
 * Ask to remove a data on target link
 * @deprecated
 * @param {string} linkId
 * @param {string} attrKey - the key of the attribute to be removed
 */
var removeLinkData = exports.removeLinkData = function removeLinkData(linkId, dataKey) {
  return {
    type: _flowdesigner.FLOWDESIGNER_LINK_REMOVE_DATA,
    linkId: linkId,
    dataKey: dataKey
  };
};

/**
 * Ask for link removal
 * @deprecated use deleteLink action
 * @param {string} linkId
 * @return {Object}
 */
var removeLink = exports.removeLink = function removeLink(linkId) {
  return {
    type: _flowdesigner.FLOWDESIGNER_LINK_REMOVE,
    linkId: linkId
  };
};