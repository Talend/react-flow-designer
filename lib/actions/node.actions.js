'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.update = exports.removeNode = exports.removeNodeData = exports.setNodeData = exports.removeNodeGraphicalAttribute = exports.setNodeGraphicalAttributes = exports.setNodeSize = exports.applyMovementTo = exports.addNode = undefined;
exports.startMoveNodeTo = startMoveNodeTo;
exports.moveNodeTo = moveNodeTo;
exports.moveNodeToEnd = moveNodeToEnd;
exports.setNodeType = setNodeType;

var _flowdesigner = require('../constants/flowdesigner.constants');

/**
 * Ask for node creation and injection into current dataflow
 * @param {string} nodeId
 * @param {{x: number, y: number}} nodePosition
 * @param {{height: number, width: number}} nodeSize
 * @param {string} nodeType
 * @param {Object} attr
 * @return {Object}
 */
var addNode = exports.addNode = function addNode(nodeId, nodeType) {
  var _ref = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {},
      _ref$data = _ref.data,
      data = _ref$data === undefined ? {} : _ref$data,
      _ref$graphicalAttribu = _ref.graphicalAttributes,
      graphicalAttributes = _ref$graphicalAttribu === undefined ? {} : _ref$graphicalAttribu;

  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_ADD,
    nodeId: nodeId,
    nodeType: nodeType,
    data: data,
    graphicalAttributes: graphicalAttributes
  };
};

/**
 * @deprecated use moveStart action
 */
function startMoveNodeTo(nodeId, nodePosition) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_MOVE_START,
    nodeId: nodeId,
    nodePosition: nodePosition
  };
}

/**
 * Ask for moving node
 * @deprecated use move action
 * @param {string} nodeId - identifier of the targeted node
 * @param {{x: number, y: number}} nodePosition - the new absolute position of the node
 * @return {Object}
 */
function moveNodeTo(nodeId, nodePosition) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_MOVE,
    nodeId: nodeId,
    nodePosition: nodePosition
  };
}

/**
 * Ask to apply the same movement to multiples nodesId
 * @deprecated
 * @param nodesId {array<string>} list of nodeId
 * @param movement {Object} relative movement to apply on all nodes
 *
 * @return {Object}
 */
var applyMovementTo = exports.applyMovementTo = function applyMovementTo(nodesId, movement) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_APPLY_MOVEMENT,
    nodesId: nodesId,
    movement: movement
  };
};

/**
 * When node movement is done
 * @deprecated use moveEnd action
 * @param {string} nodeId - identifier of the targeted node
 * @param {{x: number, y: number}} nodePosition - the new absolute position of the node
 * @return {Object}
 */
function moveNodeToEnd(nodeId, nodePosition) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_MOVE_END,
    nodeId: nodeId,
    nodePosition: nodePosition
  };
}

/**
 * set node size
 * @deprecated
 * @param {string} nodeId
 * @param {{height: number, width: number}} nodeSize
 * @return {Object}
 */
var setNodeSize = exports.setNodeSize = function setNodeSize(nodeId, nodeSize) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_SET_SIZE,
    nodeId: nodeId,
    nodeSize: nodeSize
  };
};

/**
 * Ask for node creation and injection into current dataflow
 * @deprecated
 * @param {string} nodeId
 * @param {string} nodeType
 * @return {Object}
 */
function setNodeType(nodeId, nodeType) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_SET_TYPE,
    nodeId: nodeId,
    nodeType: nodeType
  };
}

/**
 * Give the ability to a graphical attribute onto the node
 * @deprecated
 * @param {string} nodeId
 * @param {Object} graphicalAttributes
 */
var setNodeGraphicalAttributes = exports.setNodeGraphicalAttributes = function setNodeGraphicalAttributes(nodeId, graphicalAttributes) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_SET_GRAPHICAL_ATTRIBUTES,
    nodeId: nodeId,
    graphicalAttributes: graphicalAttributes
  };
};

/**
 * Ask to remove a graphical attribute on target node
 * @deprecated
 * @param {string} nodeId
 * @param {string} graphicalAttributesKey - the key of the attribute to be removed
 */
var removeNodeGraphicalAttribute = exports.removeNodeGraphicalAttribute = function removeNodeGraphicalAttribute(nodeId, graphicalAttributesKey) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_REMOVE_GRAPHICAL_ATTRIBUTES,
    nodeId: nodeId,
    graphicalAttributesKey: graphicalAttributesKey
  };
};

/**
 * Give the ability to set data onto a node
 * @deprecated
 * @param {string} nodeId
 * @param {Object} data
 * @param {boolean} bySubmit Flag to indicates that the action was triggered by a manual user action
 */
var setNodeData = exports.setNodeData = function setNodeData(nodeId, data, bySubmit) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_SET_DATA,
    nodeId: nodeId,
    data: data,
    bySubmit: bySubmit
  };
};

/**
 * Ask to remove a graphical attribute on target node
 * @deprecated
 * @param {string} nodeId
 * @param {string} dataKey - the key of the data to be removed
 */
var removeNodeData = exports.removeNodeData = function removeNodeData(nodeId, dataKey) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_REMOVE_DATA,
    nodeId: nodeId,
    dataKey: dataKey
  };
};

/**
 * Ask for removal of target node and each ports/links attached to it
 * @deprecated use deleteNode action
 * @param {string} nodeId
 */
var removeNode = exports.removeNode = function removeNode(nodeId) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_REMOVE,
    nodeId: nodeId
  };
};

var update = exports.update = function update(nodeId, node) {
  return {
    type: _flowdesigner.FLOWDESIGNER_NODE_UPDATE,
    node: node,
    nodeId: nodeId
  };
};