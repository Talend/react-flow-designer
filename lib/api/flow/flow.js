'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.updateNode = exports.deleteNode = exports.addNode = exports.isNodeExist = undefined;

var _flow = require('lodash/flow');

var _flow2 = _interopRequireDefault(_flow);

var _curry = require('lodash/curry');

var _curry2 = _interopRequireDefault(_curry);

var _immutable = require('immutable');

var _immutable2 = _interopRequireDefault(_immutable);

var _throwInDev = require('../throwInDev');

var _ = require('./..');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * check if node exist in flow
 * @param {FlowState} state
 * @param {string} nodeId
 * @return {bool} true if node exist
 */
var isNodeExist = exports.isNodeExist = (0, _curry2.default)(function (state, nodeId) {
	return state.hasIn(['nodes', nodeId]);
});

var setOut = (0, _curry2.default)(function (nodeId, state) {
	return state.setIn(['out', nodeId], new _immutable2.default.Map());
});
var deleteOut = (0, _curry2.default)(function (nodeId, state) {
	return state.deleteIn(['out', nodeId]);
});

var setIn = (0, _curry2.default)(function (nodeId, state) {
	return state.setIn(['in', nodeId], new _immutable2.default.Map());
});
var deleteIn = (0, _curry2.default)(function (nodeId, state) {
	return state.deleteIn(['in', nodeId]);
});

var setChildrens = (0, _curry2.default)(function (nodeId, state) {
	return state.setIn(['childrens', nodeId], new _immutable2.default.Map());
});
var deleteChildrens = (0, _curry2.default)(function (nodeId, state) {
	return state.deleteIn(['childrens', nodeId]);
});

var setParents = (0, _curry2.default)(function (nodeId, state) {
	return state.setIn(['parents', nodeId], new _immutable2.default.Map());
});
var deleteParents = (0, _curry2.default)(function (nodeId, state) {
	return state.deleteIn(['parents', nodeId]);
});

/**
 * add a node to the flow
 * @param {FlowState} state
 * @param {NodeRecord} node
 * @return {FlowState}
 */
var addNode = exports.addNode = (0, _curry2.default)(function (state, node) {
	if (_.Node.isNodeElseThrow(node) && !isNodeExist(state, _.Node.getId(node))) {
		var nodeId = _.Node.getId(node);
		return (0, _flow2.default)([setOut(nodeId), setIn(nodeId), setChildrens(nodeId), setParents(nodeId)])(state.setIn(['nodes', _.Node.getId(node)], node));
	}
	(0, _throwInDev.throwInDev)('Node with id = ' + _.Node.getId(node) + ', already exist, can\'t create node.');
	return state;
});

/**
 * if exist remove a node from the flow
 * @param {FlowState} state
 * @param {string} nodeId
 * @return {FlowState}
 */
var deleteNode = exports.deleteNode = (0, _curry2.default)(function (state, nodeId) {
	if (isNodeExist(nodeId)) {
		return (0, _flow2.default)([deleteOut(nodeId), deleteIn(nodeId), deleteChildrens(nodeId), deleteParents(nodeId)])(state.deleteIn(['nodes', nodeId]));
	}
	return state;
});

/**
 * update a node
 * @param {FlowState} state
 * @param {string} nodeId
 * @param {NodeRecord} node
 * @return {FlowState}
 */
var updateNode = exports.updateNode = (0, _curry2.default)(function (state, nodeId, node) {
	if (_.Node.isNodeElseThrow(node) && (isNodeExist(state, _.Node.getId(node)), isNodeExist(nodeId))) {
		if (nodeId === _.Node.getId(node)) {
			return addNode(deleteNode(state, _.Node.getId(node)), node);
		}
		return state.setIn(['nodes', nodeId], node);
	}
	return state;
});