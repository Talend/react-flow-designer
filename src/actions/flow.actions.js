import {
	FLOWDESIGNER_FLOW_ADD_ELEMENTS,
	FLOWDESIGNER_FLOW_RESET,
	FLOWDESIGNER_FLOW_LOAD,
} from '../constants/flowdesigner.constants';

/**
 * Ask to sequentially add elements to the flow, each creation should be checked against store,
 * then applied via current reducers
 *
 * @params {array} listOfActionCreation
 */
export const addFlowElements = listOfActionCreation => ({
	type: FLOWDESIGNER_FLOW_ADD_ELEMENTS,
	listOfActionCreation,
});

/**
 * ask for flow reset, emptying, nodes, links, ports collections
 */
export const resetFlow = () => ({
	type: FLOWDESIGNER_FLOW_RESET,
});

/**
 * load a new flow
 */
export const loadFlow = listOfActionCreation => ({
	type: FLOWDESIGNER_FLOW_LOAD,
	listOfActionCreation,
});
