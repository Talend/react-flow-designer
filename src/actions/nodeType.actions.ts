import { FLOWDESIGNER_NODETYPE_SET } from '../constants/flowdesigner.constants';

/**
 * Ask to set a map for nodeTypes
 * @param {Object} nodeTypes
 */
export const setNodeTypes = (nodeTypes: object) => ({
	type: FLOWDESIGNER_NODETYPE_SET,
	nodeTypes,
});

export default setNodeTypes;
