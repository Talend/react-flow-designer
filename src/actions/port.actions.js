/* @flow */

import type { PortGraphicalAttributes, Port, PortData, PortAction } from '../flow-typed';

/**
 * return an action to create a new port
 * @param {string} nodeId - identifier of the node to wich the created connector should be attached
 * @param {string} portId
 * @param {string} portType
 * @param {Object} attributes
 */
export function addPort(nodeId: string, portId: string, { data, graphicalAttributes }: {data: PortData, graphicalAttributes: PortGraphicalAttributes}): PortAction {
	return {
		type: 'FLOWDESIGNER_PORT_ADD',
		nodeId,
		portId,
		data,
		graphicalAttributes,
	};
}

export const addPorts = (nodeId: string, ports: Array<Port>): PortAction => ({
	type: 'FLOWDESIGNER_PORT_ADDS',
	nodeId,
	ports,
});

/**
 * return an action to set port attributes
 * @param {string} portId
 * @param {Object} graphicalAttributes
 */
export const setPortGraphicalAttribute = (portId: string, graphicalAttributes: PortGraphicalAttributes): PortAction => ({
	type: 'FLOWDESIGNER_PORT_SET_GRAPHICAL_ATTRIBUTES',
	portId,
	graphicalAttributes,
});

/**
 * Ask to remove an attribute on target port
 * @param {string} portId
 * @param {string} graphicalAttributesKey - the key of the attribute to be removed
 */
export const removePortGraphicalAttribute = (portId: string, graphicalAttributesKey: string): PortAction => ({
	type: 'FLOWDESIGNER_PORT_REMOVE_GRAPHICAL_ATTRIBUTES',
	portId,
	graphicalAttributesKey,
});

/**
 * return an action to set port attributes
 * @param {string} portId
 * @param {Object} graphicalAttributes
 */
export const setPortdata = (portId: string, data: Object): PortAction => ({
	type: 'FLOWDESIGNER_PORT_SET_DATA',
	portId,
	data,
});

/**
 * Ask to remove an attribute on target port
 * @param {string} portId
 * @param {string} datasKey - the key of the attribute to be removed
 */
export const removePortData = (portId: string, dataKey: string): PortAction => ({
	type: 'FLOWDESIGNER_PORT_REMOVE_DATA',
	portId,
	dataKey,
});

/**
 * return an action to remove port and all attached links
 * @param {string} portId
 */
export const removePort = (portId: string): PortAction => ({
	type: 'FLOWDESIGNER_PORT_REMOVE',
	portId,
});
