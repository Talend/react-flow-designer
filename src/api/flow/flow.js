import flow from 'lodash/flow';
import curry from 'lodash/curry';
import Immutable from 'immutable';

import { throwInDev } from '../throwInDev';
import { Node } from './..';

/**
 * check if node exist in flow
 * @param {FlowState} state
 * @param {string} nodeId
 * @return {bool} true if node exist
 */
export const isNodeExist = curry((state, nodeId) => state.hasIn(['nodes', nodeId]));

const setOut = curry((nodeId, state) => state.setIn(['out', nodeId], new Immutable.Map()));
const deleteOut = curry((nodeId, state) => state.deleteIn(['out', nodeId]));

const setIn = curry((nodeId, state) => state.setIn(['in', nodeId], new Immutable.Map()));
const deleteIn = curry((nodeId, state) => state.deleteIn(['in', nodeId]));

const setChildrens = curry((nodeId, state) =>
	state.setIn(['childrens', nodeId], new Immutable.Map()),
);
const deleteChildrens = curry((nodeId, state) => state.deleteIn(['childrens', nodeId]));

const setParents = curry((nodeId, state) => state.setIn(['parents', nodeId], new Immutable.Map()));
const deleteParents = curry((nodeId, state) => state.deleteIn(['parents', nodeId]));

/**
 * add a node to the flow
 * @param {FlowState} state
 * @param {NodeRecord} node
 * @return {FlowState}
 */
export const addNode = curry((state, node) => {
	if (Node.isNodeElseThrow(node) && !isNodeExist(state, Node.getId(node))) {
		const nodeId = Node.getId(node);
		return flow([setOut(nodeId), setIn(nodeId), setChildrens(nodeId), setParents(nodeId)])(
			state.setIn(['nodes', Node.getId(node)], node),
		);
	}
	throwInDev(`Node with id = ${Node.getId(node)}, already exist, can't create node.`);
	return state;
});

/**
 * if exist remove a node from the flow
 * @param {FlowState} state
 * @param {string} nodeId
 * @return {FlowState}
 */
export const deleteNode = curry((state, nodeId) => {
	if (isNodeExist(nodeId)) {
		return flow([
			deleteOut(nodeId),
			deleteIn(nodeId),
			deleteChildrens(nodeId),
			deleteParents(nodeId),
		])(state.deleteIn(['nodes', nodeId]));
	}
	return state;
});

/**
 * update a node
 * @param {FlowState} state
 * @param {string} nodeId
 * @param {NodeRecord} node
 * @return {FlowState}
 */
export const updateNode = curry((state, nodeId, node) => {
	if (Node.isNodeElseThrow(node) && (isNodeExist(state, Node.getId(node)), isNodeExist(nodeId))) {
		if (nodeId === Node.getId(node)) {
			return addNode(deleteNode(state, Node.getId(node)), node);
		}
		return state.setIn(['nodes', nodeId], node);
	}
	return state;
});
