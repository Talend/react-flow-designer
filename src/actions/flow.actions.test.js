import * as flowActions from './flow.actions';

describe('Check that flowActions generate proper action objects', () => {
	it('addFlowElements generate proper action object', () => {
		expect(flowActions.addFlowElements([])).toEqual({
			type: 'FLOWDESIGNER.FLOW.ADD_ELEMENTS',
			listOfActionCreation: [],
		});
	});

	it('addFlowElements generate proper action object', () => {
		expect(flowActions.addFlowElements([])).toEqual({
			type: 'FLOWDESIGNER.FLOW.ADD_ELEMENTS',
			listOfActionCreation: [],
		});
	});

	it('addFlowElements generate proper action object', () => {
		expect(flowActions.addFlowElements([])).toEqual({
			type: 'FLOWDESIGNER.FLOW.ADD_ELEMENTS',
			listOfActionCreation: [],
		});
	});
});
