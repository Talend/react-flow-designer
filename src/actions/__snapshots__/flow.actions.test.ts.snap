// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panTo should generate an action with proper coordinates 1`] = `
Object {
  "type": "FLOWDESIGNER_PAN_TO",
  "x": 100,
  "y": 100,
}
`;

exports[`panTo should return "null" if not a number 1`] = `null`;

exports[`setZoom if transform object is not well set return null 1`] = `null`;

exports[`setZoom should generate an action with proper shape 1`] = `
Object {
  "transform": Object {
    "k": 0,
    "x": 0,
    "y": 0,
  },
  "type": "FLOWDESIGNER.FLOW_SET_ZOOM",
}
`;
